odoo/fields.py

```diff
@@ -2504,6 +2504,14 @@ def create(self, record_values):
         ])
 
     def write(self, records, value):
+        if 'web_save' in records.env.context:
+            # The image options linked to the the image has to be deleted as the
+            # image is being replaced.
+            records.env['image.options'].search([
+                ['res_model', '=', records._name],
+                ['res_id', 'in', records.ids],
+                ['res_field', '=', self.name],
+            ]).unlink()
```

Comment:
Err... `image.options` only exists if `web_editor` is installed. This cannot be in the base concept of binary fields.
Supposing the patch in itself would be allowed, you can look at how it is done in https://github.com/odoo/odoo/blob/master/addons/base_sparse_field/models/fields.py.

Reply:
Maybe reacting to a `write` on `ir.attachment` would work... and at least it can be a normal override inside `web_editor`.